reposcan/common/rpm.py

```diff
@@ -10,19 +10,25 @@ class RPMParseException(Exception):
     SRPM name parsing exception.
     """
 
+NEVRA_RE = re.compile(r'(.*)-(([0-9]+):)?([^-]+)-([^-]+)\.([a-z0-9_]+)')
 
-def parse_rpm_name(name):
+def parse_rpm_name(rpm_name):
     """
     Extract components from rpm name.
     """
+    if rpm_name[-4:] == '.rpm':
+        filename = rpm_name[:-4]
+    else:
+        filename = rpm_name
```

Comment:
I purposely left rpm_name variable unchanged so the actual argument passed in would be reported if the exception is thrown.

Reply:
I was thinking about that when I was doing review, but I would still remove it as whether the function was called with or without .rpm suffix is not important as you always end up calling .match() without `.rpm` suffix thanks to previous code.
Ideally there should be zero exceptions thrown by this function and eventual lack of `.rpm` suffix which does not matter anyway does not justify the complexity of the code.

Reply:
I don't see how one "else" adds significant complexity.  If that is the issue, I can change it to:
`filename = rpm_name
if rpm_name[-4:] == '.rpm':
    filename = rpm_name[:-4]`

I think it is VERY important for the exception to return exactly the argument passed in to the method/function.  If the exception occurs and says it couldn't parse a value that is different from the value passed in, it can cause confusion for the person debugging the issue.

Reply:
Modified as above in updated PR (force pushed amended commit)